desc/magnetic_fields.py

```diff
@@ -1034,6 +1004,238 @@ def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
+            B = field.compute_magnetic_field(coords)
+
+        #########
+        # make b
+        #########
+        # we will have the rhs be 3*num_nodes in length (bc of vector B)
+
+        rhs = jnp.vstack((B[:, 0], B[:, 1], B[:, 2])).T.flatten(order="F")
+
+        #####################
+        # b is made, now do A
+        #####################
+        num_modes = 1 + (max_l + 1) * (max_m + 1) * 4
+        # TODO: if symmetric, technically only need half the modes
+        # however, the field and functions are setup to accept equal
+        # length arrays for a,b,c,d, so we will just zero out the
+        # modes that don't fit symmetry, but in future
+        # should refactor code to have a 3rd index so that
+        # we have a = V_ml0, b = V_ml1, c = V_ml2, d = V_ml3
+        # and the modes array can then be [m,l,x] where x is 0,1,2,3
+        # and we dont need to keep track of a,b,c,d separately
+
+        # TODO: technically we can drop some modes
+        # since if max_l=0, there are only ever nonzero terms for a and b
+        # and if max_m=0, there are only ever nonzero terms for a and c
+        # but since we are only fitting in a least squares sense,
+        # and max_l and max_m should probably be both nonzero anyways,
+        # this is not an issue right now
+
+        # mode numbers
+        ms = []
+        ls = []
+
+        # order of coeffs in the vector c are B0, a_ml, b_ml, c_ml, d_ml
+        a_s = []
+        b_s = []
+        c_s = []
+        d_s = []
+        zero_due_to_sym_inds = []
+        abcd_zero_due_to_sym_inds = [
+            [],
+            [],
+            [],
+            [],
+        ]  # indices that should be 0 due to symmetry
+        for l in range(max_l + 1):
+            for m in range(max_m + 1):
+                if not sym:
+                    pass  # no sym, use all coefs
+                elif l // 2 == 0:
+                    zero_due_to_sym_inds = [0, 3]  # a=d=0 for even l with sym
+                elif l // 2 == 1:
+                    zero_due_to_sym_inds = [1, 2]  # b=c=0 for odd l with sym
+                for which_coef in range(4):
+                    if which_coef == 0:
+                        a_s.append(1)
+                    elif which_coef == 1:
+                        b_s.append(1)
+                    elif which_coef == 2:
+                        c_s.append(1)
+                    elif which_coef == 3:
+                        d_s.append(1)
+                    if which_coef in zero_due_to_sym_inds:
+                        abcd_zero_due_to_sym_inds[which_coef].append(0)
+                    else:
+                        abcd_zero_due_to_sym_inds[which_coef].append(1)
+
+                ms.append(m)
+                ls.append(l)
+        for i in range(4):
+            abcd_zero_due_to_sym_inds[i] = jnp.asarray(abcd_zero_due_to_sym_inds[i])
+        assert (
+            round(jnp.sum(len(a_s) + len(b_s) + len(c_s) + len(d_s))) == num_modes - 1
```

Comment:
why round and sum?

Reply:
I think I was tired when writing this code as yea that is stupid of me to have wrote it like that...